features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Endpoints/GeneralConfiguration/EndpointSecurity.jsx

```diff
@@ -58,17 +58,23 @@ function EndpointSecurity(props) {
                 defaultMessage: 'Digest Auth',
             }),
         }];
-
     useEffect(() => {
-        setEndpointSecurityInfo(securityInfo !== null ? securityInfo : endpointSecurityInfo);
+        const tmpSecurity = {};
+        if (securityInfo !== null) {
+            const { type, username, password } = securityInfo;
+            tmpSecurity.type = type;
+            tmpSecurity.username = username;
+            tmpSecurity.password = password === '' ? '**********' : password;
```

Comment:
is there any reason we are hardcoding '**********' ? won't this be added automatically if the text field type is password ?

Reply:
If we add endpoint security and save the api, the password is not retrieved in the response. Instead, the password value is set as an empty string. So the UI shows an empty input field for password. (But user has added a password). 
To handle this (to let the users, a password is set already), we check the default password value and set a dummy value.

Reply:
let's say after saving once he clicks on save again for the second time, then his password will get updated for the empty value right? if that happens it would be an issue

Reply:
No. The value is not saved unless the user edit the password.

Reply:
@menaka, I think we could use username as a password until the user focused on the password input?